apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/UniverseDomainTest.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"):
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Storage.v1.Data;
+using System;
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests;
+
+public class UniverseDomainTest : IDisposable
+{
+    private StorageClient _storage;
+    private readonly string _universeDomain;
+    private readonly string _location;
+    private readonly string _projectId;
+    private readonly string _credsPath;
+    private readonly string _bucketName;
+    private readonly string _objectName;
+
+    public UniverseDomainTest()
+    {
+        _universeDomain = Environment.GetEnvironmentVariable("TEST_UNIVERSE_DOMAIN");
+        _projectId = Environment.GetEnvironmentVariable("TEST_UNIVERSE_PROJECT_ID");
+        _location = Environment.GetEnvironmentVariable("TEST_UNIVERSE_LOCATION");
+        _credsPath = Environment.GetEnvironmentVariable("TEST_UNIVERSE_DOMAIN_CREDENTIAL");
```

Comment:
Please check that all of these have values, fail otherwise with a descriptive message.